### PR TITLE
Bluetooth audio receiving via a2dp on esp32

### PR DESCRIPTION
**Pull request by tonyxforce**

### CODE REVIEW

**Comment by softhack007** [comment]
Wow, that's a mega huge PR 😅 will need some time for reviewing....

A few questions upfront:
- is is strictly necessary to put the complete ESP32-A2DP project into the "lib" folder? Can you use a github reference instead, and describe in your usermod readme how to add this to`lib_deps`?
- why did you make changes to other usermods? Is this strictly needed for making your new usermod work?

Due to the large amount of changes in this PR, it may take time until we manage to review everything. So please be patient.

File: usermods/a2dp/a2dp.h [hunk 1]
```diff
@@ -0,0 +1,350 @@
...
+   * loop() is called continuously. Here you can check for events, read sensors, etc.
+   *
+   * Tips:
+   * 1. You can use "if (WLED_CONNECTED)" to check for a successful network connection.
+   *    Additionally, "if (WLED_MQTT_CONNECTED)" is available to check for a connection to an MQTT broker.
+   *
+   * 2. Try to avoid using the delay() function. NEVER use delays longer than 10 milliseconds.
+   *    Instead, use a timer check as shown here.
+   */
+  void loop()
```
Comment (softhack007) [inline usermods/a2dp/a2dp.h:115]: It looks like this usermod does nothing. The usermod::loop() is empty.

Are you sure you wanted to make a PR to MoonModules ?

**Review by softhack007 — CHANGES_REQUESTED** [review]
the usermod currently adds too many files, contains a lot of unrelated changes; and above all  it cannot work (usermod main loop is empty).

I will not merge this.

**Comment by softhack007** [comment]
Hi @tonyxforce, I'm closing this PR because it looks to me that you opened it by mistake.

The idea is interesting, but you have a long way ahead of you to get this working.
I also have some doubts that the concept of running everything - a bluetooth a2dp decoder, wifi, webserver, and WLED effects - on one MCU will work at all.  It might just be too much workload for a single esp board.


Maybe the smarter solution will be to use two boards:
1) one board running WLEDMM with standard audioreactive
2) a second board running the bluetooth a2dp part, which sends UDP sound sync packets to the board running WLEDMM

cheers, Frank

**Comment by tonyxforce** [comment]
I've tested this code without WLED and it worked perfectly with an empty loop i'm currently just trying my best to merge WLED an this code

About the package installation tip, can you give me some more specific tips about how could i do that?

**Comment by tonyxforce** [comment]
The reason of the so many commits and the changes in other files is that i originally intended my repo to be completely separated from wled and delete everything from it i didn't personally need in my specific hardware and halfway through my coding i realized WLED could benefit if i made it a usermod so i put everything back and started working on it after i created this pull request

**Comment by softhack007** [comment]
> About the package installation tip, can you give me some more specific tips about how could i do that?

It's very simple to use a library from another repo on github.
You just add it to `lib_deps` in you custom platformio build environment. Similar to this

https://github.com/MoonModules/WLED/blob/f7df60932ab67c8f42a480ce4c6e8a40965355cc/platformio.ini#L308-L315


The format is
* user/repo @ release (makuna/NeoPixelBus @ ^2.7.5)
* fullURL.git # branch (https://github.com/softhack007/LITTLEFS-threadsafe.git#master)
However the first format is generally preferred, because it links to a specific release so the build will still work in a few years.

**Comment by tonyxforce** [comment]
Oh thanks! I didn't know you can do that with pio! Tomorrow i'll fix the issues and maybe recreate(delete/create new) repository for it to be a fresh start

**Pull request closed without merging**